cloudbuild.yaml

```diff
@@ -0,0 +1,19 @@
+# steps:
+# - name: "gcr.io/cloud-builders/mvn"
+#   args:
+#   - package
+# artifacts:
+#   objects:
+#   // This is using sources from teh Google Cloud Source Repository
+#     location: "gs://${_BUCKET}/"
+#     paths:
+#     - "target/${_TARGET}"
+
+steps:
+- id: 'Staging the app using mvn appengine plugin'
+  name: 'gcr.io/cloud-builders/mvn'
+  args: ['package', 'appengine:stage']
+- id: "Deploy to app engine using gcloud image"
+  name: 'gcr.io/cloud-builders/gcloud'
+  args: ['app', 'deploy', 'codeu-spring-2019-team-9\cloudbuild.yaml']
```

Comment:
Minor note for future reference: the *.yaml file that is expected in a call to `gcloud app deploy` is not the Cloud Build configuration, but rather the configuration for App Engine (that YAML file is automatically generated during the build process from `appengine-web.xml`, `web.xml`, and the other build artifacts).

You can see what such a file looks like in the Python App Engine case here:
https://cloud.google.com/appengine/docs/standard/python/config/appref

The Java App Engine environment uses *.xml in order to be idiomatic (Java uses XML more than YAML), but then that configuration gets converted into YAML as in the Python case.

Based on the documentation for the `appengine:stage` command here:
https://cloud.google.com/appengine/docs/standard/java/tools/maven-reference#appenginestage

... it would appear that the *.yaml file might be output in a subdirectory of `target/appengine-staging`, but it shouldn't be necessary to figure out the details of the mechanics since `appengine:deploy` will do all of this under the hood and magically invoke `gcloud app deploy` with the correct parameters.

Reply:
> 
> 
> Can you add a comment that explains why you do this in multiple steps? That is, why not:
> 
> ```
> steps:
> # Builds the application and deploys it to App Engine using Maven.
> - id: 'build_and_deploy'
>   name: 'gcr.io/cloud-builders/mvn'
>   args: ['package', 'appengine:deploy']
>   env:
>     - "GOOGLE_CLOUD_PROJECT=sp19-codeu-9-3673"
> ```
> 
> I can understand splitting it into multiple steps if, for example, you did some sort of validation step after the build but before the deployment that could be used to prevent the deployment from happening. But since there isn't such a step, it isn't clear to me why you've broken it down into the separate steps like this.

So I tried this approach that was mentioned and wasn't able to figure out how to fix it, was still getting build errors.

-----------------------------------------------------------------------------------------------------------
I haven't been able test the cloudbuild locally, but I created a new app engine account to test out the cloudbuild.yaml. 

I changed the yaml from
```codeu-spring-2019-team-9\cloudbuild.yaml``` -> ```src/main/webapp/WEB-INF/appengine-web.xml```

The only thing is that it does take awhile to build, avg 1 min 30 sec. 

I will see if I am able to speed up the process by "caching directories with Google Cloud Storage"